src/app/wallet/wallet/history/history.component.ts

```diff
@@ -1,28 +1,88 @@
-import { Component } from '@angular/core';
-
-import { TransactionCategory } from '../shared/transaction.model';
+import { Component, OnInit, ViewChild } from '@angular/core';
 
 @Component({
   selector: 'app-history',
   templateUrl: './history.component.html',
   styleUrls: ['./history.component.scss']
 })
-export class HistoryComponent {
 
-  category: TransactionCategory = 'all';
-  tabsTtitles: Array<any> = ['all', 'send', 'receive', 'stake', 'orphaned Stake'];
+export class HistoryComponent implements OnInit {
+
+  // @ViewChild('filterList') filterList: any;
+  @ViewChild('transactions') transactions: any;
+
+  categories: Array<any> = [
+    { title: 'All transactions', value: 'all'               },
+    { title: 'Send',             value: 'send'              },
+    { title: 'Receive',          value: 'receive'           },
+    { title: 'Stake',            value: 'stake'             },
+    { title: 'Balance transfer', value: 'internal_transfer' },
+    // { title: 'Immature',         value: 'immature'          },
+    // { title: 'Coinbase',         value: 'coinbase'          },
+    // { title: 'Orphan',           value: 'orphan'            },
+    // { title: 'Orphaned stake',   value: 'orphaned_stake'    },
+  ];
+
+  sortings: Array<any> = [
+    { title: 'By time',                  value: 'time'          },
+    { title: 'By amount',                value: 'amount'        },
+    { title: 'By address',               value: 'address'       },
+    { title: 'By category',              value: 'category'      },
+    { title: 'By confirmations',         value: 'confirmations' },
+    { title: 'By transaction ID (txid)', value: 'txid'          }
+  ];
+
+  types: Array<any> = [
+    { title: 'All types', value: 'all'      },
+    { title: 'Standard',  value: 'standard' },
+    { title: 'Anonymous', value: 'anon'     },
+    { title: 'Blind',     value: 'blind'    },
+  ];
+
+  filters: any = {
+    category: undefined,
+    search:   undefined,
+    sort:     undefined,
+    type:     undefined
+  };
+
+  // default tab is zero
+  public selectedTab: number = 0;
```

Comment:
this does not seem to change the tab. try with other numbers to test

Reply:
you can change no zero to 1 tab would change.